sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -255,12 +255,28 @@ trait JoinSelectionHelper {
     val buildLeft = if (hintOnly) {
       hintToShuffleHashJoinLeft(hint)
     } else {
-      canBuildLocalHashMapBySize(left, conf) && muchSmaller(left, right)
+      if (hintToPreferShuffleHashJoinLeft(hint)) {
```

Comment:
what about the case when both left and right sides meet requirement of AQE shuffled hash join (i.e. each partition size < threshold) ? Here we always build left. Shall we build right in case right side is smaller than left side?

Reply:
Yeah,  we have already chose a smaller one since the `LogicalQueryStage` has override the `computeStats`. And see the code in `joins.getBuildSide`:
```
private def getBuildSide(
      canBuildLeft: Boolean,
      canBuildRight: Boolean,
      left: LogicalPlan,
      right: LogicalPlan): Option[BuildSide] = {
    if (canBuildLeft && canBuildRight) {
      // returns the smaller side base on its estimated physical size, if we want to build the
      // both sides.
      Some(getSmallerSide(left, right))
    } else if (canBuildLeft) {
      Some(BuildLeft)
    } else if (canBuildRight) {
      Some(BuildRight)
    } else {
      None
    }
  }
```

Reply:
Add an build side check if you worry about this.  [c747a23](https://github.com/apache/spark/pull/32550/commits/c747a23be0e59bcb7edb6e9bb377d2aed7248c96)